packages/python-google-compute-engine/google_compute_engine/networking/network_setup/network_setup.py

```diff
@@ -43,23 +43,45 @@ def __init__(self, dhclient_script=None, dhcp_command=None, debug=False):
     self.logger = logger.Logger(
         name='network-setup', debug=debug, facility=facility)
     self.distro_utils = distro_utils.Utils(debug=debug)
+    self.ipv6_initialized = False
 
   def EnableIpv6(self, interfaces):
     """Enable IPv6 on the list of network interfaces.
 
     Args:
       interfaces: list of string, the output device names for enabling IPv6.
     """
-    if not interfaces or set(interfaces) == self.ipv6_interfaces:
+    if not interfaces or self.ipv6_interfaces == set(interfaces):
```

Comment:
is this checking whether the requested interfaces have already been enabled?

Reply:
that is correct.